src/main/kotlin/calculator/StringDelimiterSeparator.kt

```diff
@@ -0,0 +1,36 @@
+package calculator
+
+@JvmInline
+value class StringDelimiterSeparator(private val delimiterRegex: Regex) {
+
+    init {
+        val pattern = delimiterRegex.pattern
+        require(pattern.contains("<$DELIMITER_GROUP_NAME>") and pattern.contains("<$STRINGS_GROUP_NAME>")) { "$DELIMITER_GROUP_NAME and $STRINGS_GROUP_NAME must be included in regex pattern. but provided pattern(`$delimiterRegex`)" }
+    }
+
+    infix fun split(string: String): Collection<String>? {
```

Comment:
```suggestion
    infix fun split(string: String): Collection<String> {
```
List, Map, Set과 같은 컬렉션은 null 대신 빈컬렉션으로도 유효하지 않다는 것을 표현해보면 어떨까요?